src/gurt/telemetry.c

```diff
@@ -837,14 +857,24 @@ d_tm_init(int id, uint64_t mem_size, int flags)
 void
 d_tm_fini(void)
 {
-	bool	destroy_shmem = false;
+	bool	destroy_shmem = true;
 
 	if (tm_shmem.ctx == NULL)
 		goto out;
 
-	if (!tm_shmem.retain)
-		destroy_shmem = true;
+	if (tm_shmem.retain)
+		destroy_shmem = false;
+
+	if (tm_shmem.retain_non_empty) {
+		struct d_tm_node_t *root;
+
+		root = d_tm_get_root(tm_shmem.ctx);
+		if (root->dtn_child != NULL)
+			destroy_shmem = false;
+	}
 
+	D_INFO("Delete share memory for id %u destory %s\n", tm_shmem.id,
```

Comment:
There's a DF_BOOL macro to use here rather than yes/no trigraph.

Reply:
sure